types/validator_set_test.go

```diff
@@ -599,3 +599,217 @@ func TestValidatorSetVerifyCommit(t *testing.T) {
+	v2d := newValidator([]byte("v2"), 0)
+	v2m := newValidator([]byte("v2"), 10)
+	valList = []*Validator{v2d, v2m}
+	valsc = vals.Copy()
+	assert.NotNil(t, vals.UpdateWithChangeSet(valList))
+	assert.Nil(t, verifyValidatorSetEquiv(vals, valsc))
+
+	// Verify negative voting power is caught and vals is not modified
+	v1 = newValidator([]byte("v3"), -10)
+	valList = []*Validator{v1}
+	valsc = vals.Copy()
+	assert.NotNil(t, vals.UpdateWithChangeSet(valList))
+	assert.Nil(t, verifyValidatorSetEquiv(vals, valsc))
+
+	// Verify deleting unexisting validator is caught and vals is not modified
+	v1 = newValidator([]byte("v7"), 0)
+	valList = []*Validator{v1}
+	valsc = vals.Copy()
+	assert.NotNil(t, vals.UpdateWithChangeSet(valList))
+	assert.Nil(t, verifyValidatorSetEquiv(vals, valsc))
+
+	// Change priorities
+	v1 = newValidator([]byte("v1"), 10)
+	v2 = newValidator([]byte("v2"), 10)
+	valList = []*Validator{v1, v2}
+	assert.Nil(t, vals.UpdateWithChangeSet(valList))
+	assert.Nil(t, verifyValidatorSet(vals))
+
+	// Add new validators
+	v3 := newValidator([]byte("v3"), 200)
+	v4 := newValidator([]byte("v4"), 200)
+	newVals := []*Validator{v3, v4}
+	assert.Nil(t, vals.UpdateWithChangeSet(newVals))
+	assert.Nil(t, verifyValidatorSet(vals))
+
+	// Delete validators
+	v3 = newValidator([]byte("v3"), 0)
+	v4 = newValidator([]byte("v4"), 0)
+	delVals := []*Validator{v3, v4}
+	assert.Nil(t, vals.UpdateWithChangeSet(delVals))
+	assert.Nil(t, verifyValidatorSet(vals))
+
+	// Verify order does not matter when adding new validators
+	vals1 := vals.Copy()
+	vals2 := vals.Copy()
+	v3 = newValidator([]byte("v3"), 333)
+	v4 = newValidator([]byte("v4"), 444)
+
+	newVals = []*Validator{v3, v4}
+	assert.Nil(t, vals1.UpdateWithChangeSet(newVals))
+	assert.Nil(t, verifyValidatorSet(vals1))
+
+	newVals = []*Validator{v4, v3}
+	assert.Nil(t, vals2.UpdateWithChangeSet(newVals))
+	assert.Nil(t, verifyValidatorSet(vals2))
+
+	assert.Nil(t, verifyValidatorSetEquiv(vals1, vals2))
+
+	// Verify order does not matter when modifying validators
+	vals1 = vals.Copy()
+	vals2 = vals.Copy()
+	v3 = newValidator([]byte("v3"), 33)
+	v4 = newValidator([]byte("v4"), 44)
+
+	chVals := []*Validator{v3, v4}
+	assert.Nil(t, vals1.UpdateWithChangeSet(chVals))
+	assert.Nil(t, verifyValidatorSet(vals1))
+
+	chVals = []*Validator{v4, v3}
+	assert.Nil(t, vals2.UpdateWithChangeSet(chVals))
+	assert.Nil(t, verifyValidatorSet(vals2))
+
+	assert.Nil(t, verifyValidatorSetEquiv(vals1, vals2))
+
+	// Verify order does not matter when updates contain new and changed validators
+	vals1 = vals.Copy()
+	vals2 = vals.Copy()
+	v3 = newValidator([]byte("v3"), 3)
+	v5 := newValidator([]byte("v4"), 55)
+
+	chVals = []*Validator{v3, v5}
+	assert.Nil(t, vals1.UpdateWithChangeSet(chVals))
+	assert.Nil(t, verifyValidatorSet(vals1))
+
+	chVals = []*Validator{v5, v3}
+	assert.Nil(t, vals2.UpdateWithChangeSet(chVals))
+	assert.Nil(t, verifyValidatorSet(vals2))
```

Comment:
This could be changed to: `assert.NoError`. In that way the reader immediately would notice, that  `verifyValidatorSet` returns an error.

Reply:
Alternatively (as this is a test), you could also pass around `t` and do the asserts inside of `verifyValidatorSet(t *testing.T, vals2)` etc.

Reply:
Will try both suggestions, thanks!